src/spetlr/deltaspec/DeltaTableSpecDifference.py

```diff
@@ -0,0 +1,356 @@
+            if len(fields_to_remove) > 1:
+                statement += "S"
+            statement += " (" + ", ".join(fields_to_remove) + ")"
+
+            statements.append(statement)
+
+        if fields_to_add:
+            if not allow_new_columns:
+                raise TableSpectNotEnforcable(
+                    "Cannot make the storage match without allowing new columns"
+                )
+
+        if fields_to_add:
+            statement = f"ALTER TABLE {self.base.name} ADD COLUMN"
+            if len(fields_to_add) == 1:
+                statement += (
+                    " ("
+                    + SchemaManager().struct_to_sql(StructType(fields_to_add)).strip()
+                    + ")"
+                )
+            else:
+                statement += "S"
+                statement += (
+                    " (/n"
+                    + SchemaManager().struct_to_sql(StructType(fields_to_add))
+                    + "\n)"
+                )
+
+            statements.append(statement)
+
+        for key, newValue in alter_nullability:
+            statements.append(
+                f"ALTER TABLE {self.base.name} ALTER COLUMN {key} "
+                + ("SET" if newValue else "DROP")
+                + " NOT NULL"
+            )
+
+        for key, newComment in alter_comment:
+            statements.append(
+                f"ALTER TABLE {self.base.name} ALTER COLUMN "
+                f"{key} COMMENT {json.dumps(newComment)}"
+            )
+
+        # Finally address the possible change of order.
+
+        intermediate_target_order = copy.copy(base_order)
+        for field in fields_to_add:
+            intermediate_target_order.append(field.name)
+
+        # now the intermediate_target_order always contains
+        # the same fields in the same order as the table for now.
+        first = target_order[0]
+        if intermediate_target_order[0] != first:
+            intermediate_target_order.pop(intermediate_target_order.index(first))
+            intermediate_target_order.insert(0, first)
+            statements.append(
+                f"ALTER TABLE {self.base.name} ALTER COLUMN {first} FIRST"
+            )
+
+        for i, key in enumerate(target_order[1:], 1):
+            if intermediate_target_order[i] != key:
+                intermediate_target_order.pop(intermediate_target_order.index(key))
+                intermediate_target_order.insert(i, key)
+                statements.append(
+                    f"ALTER TABLE {self.base.name} ALTER COLUMN {key} AFTER "
+                    + target_order[i - 1]
+                )
+
+        # possible differences:
+        # - different capitalization -- Not handled. Treated as new column
+        # - different nullability -- Done
+        # - different struct members (new, dropped)
+        #           -- Not handled. Treated as new column
+        # - different comment
+        # - different order
+
+        # Todo: FIX THEM ALL
```

Comment:
WHat is the status? are "different comment" and "different order" done?

Reply:
comment removed. all handled.